presto-orc/src/main/java/com/facebook/presto/orc/metadata/StripeMetaCache.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.orc.metadata;
+
+import com.facebook.airlift.log.Logger;
+import com.google.common.collect.ImmutableList;
+import io.airlift.slice.Slice;
+
+import java.util.List;
+import java.util.Optional;
+
+import static com.facebook.presto.orc.metadata.StripeMetaCacheMode.FOOTER;
+import static com.facebook.presto.orc.metadata.StripeMetaCacheMode.INDEX;
+import static com.facebook.presto.orc.metadata.StripeMetaCacheMode.INDEX_AND_FOOTER;
+import static com.facebook.presto.orc.metadata.StripeMetaCacheMode.NONE;
+import static java.util.Objects.requireNonNull;
+
+public class StripeMetaCache
+{
+    private static final Logger log = Logger.get(StripeMetaCache.class);
+    private final List<Integer> cacheOffsets;
+    private final StripeMetaCacheMode cacheMode;
+    private final Slice data;
+
+    public StripeMetaCache(List<Integer> offsets, StripeMetaCacheMode mode, Slice data)
+    {
+        this.cacheOffsets = ImmutableList.copyOf(requireNonNull(offsets, "offsets is null"));
+        this.data = requireNonNull(data, "data is null");
+        this.cacheMode = filterMode(requireNonNull(mode, "mode is null"));
+    }
+
+    // work only with supported modes
+    private StripeMetaCacheMode filterMode(StripeMetaCacheMode mode)
```

Comment:
is this to protect against a case that new cache mode is added?

Reply:
Yes. In this case the offset selection logic would change.